### ISSUE TITLE
[Bug]: Can't pass less args to an untyped function than originally passed (minus extra args)

### ISSUE THREAD

**Issue by matthewnessworthy** [DataDog/dd-trace-php#2174]
### Bug report

```
Can't pass less args to an untyped function than originally passed (minus extra args) in /opt/datadog-php/dd-trace-sources/bridge/_generated_integrations.php:2054
```

### PHP version

8.1

### Tracer or profiler version

0.90.0

### Upgrading from

0.89.1

**Comment by PROFeNoM** [comment]
Hi Matthew,

Thank you for providing the debug log. I see why you are getting this error message, and I'm currently trying to reproduce the issue. However, it led to nothing currently...

Can I ask you some questions for reproducibility purposes?
- I imagine you are getting this error message from manual log calls, right? Would it be possible for you to provide what such a call of yours looks like?
- What logging library are you using?

**Comment by matthewnessworthy** [comment]
Hi @PROFeNoM,
We're running Laravel 9 & 10 (different apps, same error), with the built-in MonoLog error logging library.

Here are the only snippets in the app where we interact with DD directly

```
        if (
            ($rootScope = \DDTrace\GlobalTracer::get()->getRootScope())
            && ($rootSpan = $rootScope->getSpan())
        ) {
            $rootSpan->setTag('domain', (string) $domain);
        }
```

```
        if (
            ($rootScope = \DDTrace\GlobalTracer::get()->getRootScope())
            && ($rootSpan = $rootScope->getSpan())
        ) {
            $rootSpan->setMetric('php.mem.peak_usage_bytes', memory_get_peak_usage(false));
            $rootSpan->setMetric('php.mem.peak_real_usage_bytes', memory_get_peak_usage(true));
        }
```

```
        if (
            ($rootScope = GlobalTracer::get()->getRootScope())
            && ($rootSpan = $rootScope->getSpan())
        ) {
            $rootSpan->setTag('domain', (string) tenant('slug'));

            $rootSpan->setTag('http.referer', $request->header('HTTP_REFERER'));
            $rootSpan->setTag('http.request_id', $request->server('HTTP_X_REQUEST_ID'));
            $rootSpan->setTag('http.useragent', $request->userAgent());
        }
```

```
        try {
            if (function_exists('\DDTrace\current_context')) {
                // @see https://docs.datadoghq.com/logs/log_collection/php/?tab=phpmonolog#laravel
                $context = \DDTrace\current_context();
                $record['dd']['trace_id'] = $context['trace_id'];
                $record['dd']['span_id'] = $context['span_id'];
            }
        } catch (Throwable $e) {
            $record['dd']['context_error'] = $e->getMessage();
        }
```

**Comment by PROFeNoM** [comment]
Hey @matthewnessworthy!

It looks like @bwoebi may have found out about the issue you are having. Would it be possible for you to try out the corresponding [artifact](https://output.circle-artifacts.com/output/job/1270cfea-7cb3-4dea-b9fb-703f9414697c/artifacts/0/datadog-setup.php) and see if it solves the problem? Let us know if there's anything else we can do to assist you.

🔧 Just in case you are not familiar with using the artifact, you simply need to replace the link of the [installer](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/php/?tab=containers#install-the-extension) (`datadog-setup.php`) to be the one provided above (generated [here](https://app.circleci.com/pipelines/github/DataDog/dd-trace-php/12096/workflows/dc6b3ccc-49e2-4e46-9b5d-925134d37063/jobs/2636061/artifacts)).

**Comment by jpasquers** [comment]
Able to reproduce this on php `8.1` and dd-trace-php `0.90.0`. Our usage is as simple as:

```
        dd_trace_method(
            CController::class,
            'getViewFile',
            function ($span, array $args, string $result, $exception) {
                $span->service = 'monolith-app';
                $span->name = 'getView.file';
                $span->resource = $result;
            },
        );
```

**Comment by jpasquers** [comment]
Following up here, we isolated the issue to our usage of JIT with php8. Particularly, this exact addition to `php.ini` caused this log to be spammed:

```
opcache.jit_buffer_size=512M
opcache.jit=1255
```

I'm not sure if this helps @bwoebi or @PROFeNoM

**Comment by PROFeNoM** [comment]
Hi @jpasquers 

Thanks for providing some more detailed information on this issue. However, I'm wondering some little things
- Do you still have this issue when using the artifact above?
- Is this issue disappearing when turning off the Logs integration (`DD_TRACE_LOGS_ENABLED=0`)? It should.
- Do you have the output of `phpinfo()`? 

So far, I've been trying to reproduce the behavior by experimenting with `\DDTrace\HookData::overrideArguments` locally and with the Logs Integration on a test app (Laravel using Monolog, PHP 8.1), adding these in my inis
```
opcache.enable=1
opcache.enable_cli=1
opcache.jit_buffer_size=512M
opcache.jit=1255
```

**Comment by PROFeNoM** [comment]
Hi! 

Good news, we've been able to [reproduce](https://github.com/DataDog/dd-trace-php/compare/bob/better-override-args-check) this issue with the additional information you gave us, which confirms the artifact addresses the issue 😃 

It will be fixed (#2227) in the next release, but in the meantime, you should be able to use the artifact above :+1:
